cli/command/container/create.go

```diff
@@ -160,16 +159,21 @@ func createContainer(ctx context.Context, dockerCli *command.DockerCli, config *
 	}
 
 	var trustedRef reference.Canonical
-	_, ref, err := reference.ParseIDOrReference(config.Image)
+	var namedRef reference.Named
+
+	ref, err := reference.ParseAnyReference(config.Image)
 	if err != nil {
 		return nil, err
 	}
-	if ref != nil {
-		ref = reference.WithDefaultTag(ref)
+	if named, ok := ref.(reference.NormalizedNamed); ok {
+		namedRef = named.Familiar()
+		if reference.IsNameOnly(namedRef) {
+			namedRef = reference.EnsureTagged(namedRef)
+		}
 
-		if ref, ok := ref.(reference.NamedTagged); ok && command.IsTrusted() {
+		if taggedRef, ok := namedRef.(reference.NamedTagged); ok && command.IsTrusted() {
 			var err error
-			trustedRef, err = image.TrustedReference(ctx, dockerCli, ref, nil)
+			trustedRef, err = image.TrustedReference(ctx, dockerCli, taggedRef, nil)
```

Comment:
Probably, reason I did it there was because there are 2 spots further on that rely on the familiar reference. However I think you are right in that the normalized form should be passed into `TrustedReference` since it resolves repository info.